test/queue/queueservice_test.js

```diff
@@ -164,4 +171,70 @@ suite('queue/QueueService', function() {
     debug("pending message count: %j", count);
     assert(typeof(count) === 'number', "Expected count as number!");
   });
+
+  test("deleteUnusedWorkerQueues (can delete queues)", async () => {
+    // 11 days into the future, so we'll delete all queues (yay)
+    let now = new Date(Date.now() + 11 * 24 * 60 * 60 * 1000);
+    let deleted = await queueService.deleteUnusedWorkerQueues(now);
+    // There should always be at least one queue, because test cases above
+    // will have created one we can delete...
+    assume(deleted).is.atleast(1);
+  });
+
+
+  test("deleteUnusedWorkerQueues (respects meta-data)", async () => {
+    // Ensure a queue with updated meta-data exists
+    let provisionerId = slugid.v4();
+    let workerType = slugid.v4();
+    let queueName = await queueService.ensurePendingQueue(
+      provisionerId, workerType
+    );
+
+    // Let's delete from now and check that we didn't delete queue just created
+    await queueService.deleteUnusedWorkerQueues();
+
+    // Get meta-data, this will fail if the queue was deleted
+    await queueService.client.getMetadata(queueName);
```

Comment:
Sure, if I'm willing to pay the cost when/if this test was to fail... I suspect it won't fail anytime soon.